core/jazz_slack-event-handler/index.js

```diff
@@ -0,0 +1,311 @@
+            var err = exportable.handleError(failureCodes.PR_ERROR_1.code, "Validation error while processing event for service");
+            return reject(err);
+        }
+        exportable.getServiceDetails(payload, configData, authToken)
+            .then(result => { return notifySlackChannel(result, payload, configData, authToken) })
+            .then(result => { resolve(result) })
+            .catch(error => {
+                return reject(error);
+            });
+    });
+}
+
+function getSvcPayload(method, payload, apiEndpoint, authToken) {
+    var svcPayload = {
+        headers: {
+            'content-type': "application/json",
+            'authorization': authToken
+        },
+        rejectUnauthorized: false
+    }
+
+    svcPayload.uri = apiEndpoint;
+    svcPayload.method = method;
+    if (payload) {
+        svcPayload.json = payload;
+    }
+    logger.info("payload :" + JSON.stringify(svcPayload));
+    return svcPayload;
+};
+
+function processRequest(svcPayload) {
+    return new Promise((resolve, reject) => {
+        request(svcPayload, function (error, response, body) {
+            if (response.statusCode === 200 && body) {
+                return resolve(body);
+            } else {
+                logger.error("Error processing request: " + JSON.stringify(response));                
+                return reject(error);
+            }
+        });
+    });
+};
+
+function getServiceDetails(eventPayload, configData, authToken) {
+    return new Promise((resolve, reject) => {
+        var service_id = eventPayload.SERVICE_ID.S;
+        var apiEndpoint = configData.SERVICE_API_URL + configData.SERVICE_API_RESOURCE + "/" + service_id;
+        var svcPayload = exportable.getSvcPayload("GET", null, apiEndpoint, authToken);
+
+        exportable.processRequest(svcPayload)
+            .then(result => { return resolve(result); })
+            .catch(err => {
+                logger.error("getServiceDetails failed: " + JSON.stringify(err));
+                var error = exportable.handleError(failureCodes.PR_ERROR_1.code, failureCodes.PR_ERROR_1.message);
+                return reject(error);
+            });
+    });
+};
+
+function notifySlackChannel(result, payload, configData, authToken) {
+    return new Promise((resolve, reject) => {
+        var output = JSON.parse(result);
+        if (!output.data) {
+            logger.error("Service details not foound in service catalog");
+            var error = exportable.handleError(failureCodes.PR_ERROR_1.code, failureCodes.PR_ERROR_1.message);
+            return reject(error);
+        }
+        var serviceDetails = output.data;
+        logger.info("service details : " + JSON.stringify(serviceDetails));
+        var slackChannel = serviceDetails.slack_channel;
+        if (!slackChannel) {
+            logger.error("Slack channel not found");
+            var error = exportable.handleError(failureCodes.PR_ERROR_5.code, failureCodes.PR_ERROR_5.message);
```

Comment:
you have multiple declarations for `error` here (line 220). use `let` as its scope is local.